phoenix-core/src/main/java/org/apache/phoenix/util/QueryUtil.java

```diff
@@ -769,6 +772,42 @@ public static PreparedStatement getTablesStmt(PhoenixConnection connection, Stri
         return stmt;
     }
 
+    /**
+     * Util that generates a PreparedStatement against syscat to get the table listing in a given schema.
+     */
+    public static PreparedStatement getShowCreateStatementStmt(PhoenixConnection connection, String catalog, TableName tn) throws SQLException {
+
+        String output;
+        SchemaProcessor processor = new SchemaExtractionProcessor(null,
+                connection.unwrap(PhoenixConnection.class).getQueryServices().getConfiguration(),
+                tn.getSchemaName() == null ? null : "\"" + tn.getSchemaName()+ "\"",
+                "\"" + tn.getTableName() + "\"");
+        try {
+            output = processor.process();
+        } catch (Exception e) {
+            LOGGER.error(e.getStackTrace().toString());
+            throw new SQLException(e.getMessage());
+        }
+
+        String stmtType;
```

Comment:
Not sure we need to adjust the column name.
"CREATE STATEMENT" is probably enough.